gmso/core/topology.py

```diff
@@ -503,7 +503,7 @@ def add_connection(self, connection, update_types=True):
             is in the topology
         """
         # Check if an equivalent connection is in the topology
-        equivalent_members = connection._equivalent_members_hash()
+        equivalent_members = connection.equivalent_members()
         if equivalent_members in self._unique_connections:
```

Comment:
So how does this `in` comparison work now? Now `equivalent_members` isn't a hash but a set of lists containing possible members. And `_unique_connections` is a dictionary of equivalent_members with value of relevant connections. This looks like it will work fine, as long as it doesn't take too long to check for these equivalent members in the unique connections. I'll double check the speed here to make sure we aren't spending too long on these checks.